src/fparser/two/tests/fortran2003/test_format_item_r1003.py

```diff
@@ -0,0 +1,172 @@
+
+
+def test_data_edit_descriptor(f2003_create):
+    '''Check that basic format specifications are parsed correctly for a
+    data edit descriptor. The description is tested in more detail by
+    the associated class.
+
+    '''
+    # No R
+    for my_input in ["F2.2", " F2.2 "]:
+        ast = Format_Item(my_input)
+        assert my_input.replace(" ", "") in str(ast)
+        assert repr(ast) == ("Format_Item(None, Data_Edit_Desc_C1002('F',"
+                             " Digit_String('2', None), Int_Literal_Constant("
+                             "'2', None), None))")
+    # R
+    for my_input in ["2F2.2", " 2 F2.2 "]:
+        ast = Format_Item(my_input)
+        assert my_input.replace(" ", "") in str(ast)
+        assert repr(ast) == ("Format_Item(Digit_String('2', None), Data_Edit"
+                             "_Desc_C1002('F', Digit_String('2', None), "
+                             "Int_Literal_Constant('2', None), None))")
+    # Multi-R
+    for my_input in ["22F2.2", " 2 2 F2.2 "]:
+        ast = Format_Item(my_input)
+        assert my_input.replace(" ", "") in str(ast)
+        assert repr(ast) == ("Format_Item(Digit_String('22', None), Data_Edit"
+                             "_Desc_C1002('F', Digit_String('2', None), "
+                             "Int_Literal_Constant('2', None), None))")
+
+
+def test_control_edit_descriptor(f2003_create):
+    '''Check that basic format specifications are parsed correctly for a
+    control edit descriptor. The description is tested in more detail by
+    the associated class.
+
+    '''
+    for my_input in ["2P", " 2P "]:
+        ast = Format_Item(my_input)
+        assert my_input.strip() in str(ast)
+        assert repr(ast) == ("Control_Edit_Desc(Signed_Int_Literal_Constant("
+                             "'2', None), 'P')")
+
+
+def test_char_edit_descriptor(f2003_create):
+    '''Check that basic format specifications are parsed correctly for a
+    char string edit descriptor. The description is tested in more
+    detail by the associated class.
+
+    '''
+    for my_input in ["'hello'", " 'hello' "]:
+        ast = Format_Item(my_input)
+        assert my_input.strip() in str(ast)
+        assert repr(ast) == ("Char_Literal_Constant(\"'hello'\", None)")
+
+
+def test_format_list_descriptor(f2003_create):
+    '''Check that basic format specifications are parsed correctly for a
+    format item list descriptor. The description is tested in more
+    detail by the associated class.
+
+    '''
+    # No R
+    for my_input in ["(F2.2)", " (F2.2) ", " ( F2.2 ) "]:
+        ast = Format_Item(my_input)
+        assert my_input.replace(" ", "") in str(ast)
+        assert repr(ast) == ("Format_Item(None, Format_Item(None, Data_Edit"
+                             "_Desc_C1002('F', Digit_String('2', None), Int"
+                             "_Literal_Constant('2', None), None)))")
+    # R
+    for my_input in ["2(F2.2)", " 2 (F2.2) "," 2 ( F2.2 ) "]:
```

Comment:
pylint complains about bad whitespace on this line.

Reply:
Oops, missing space. I was sure I had checked these files with pylint.